site/src/components/roadmap/Roadmap.tsx

```diff
@@ -0,0 +1,246 @@
+import { ReactNode, SetStateAction, useEffect, useState } from "react";
+import {
+  GridLayout,
+  Card,
+  H1,
+  CardProps,
+  Button,
+  Input,
+  H3,
+  Link,
+  Banner,
+  BannerContent,
+} from "@salt-ds/core";
+import styles from "./style.module.css";
+import {
+  ProgressInprogressIcon,
+  ProgressPendingIcon,
+  SearchIcon,
+} from "@salt-ds/icons";
+import { Heading3 } from "../mdx/h3";
+
+type RoadmapProps = { title: string; children: ReactNode };
+
+interface RoadmapData {
+  id: string;
+  startDate: Date;
+  targetDate: Date;
+  issueUrl: string;
+  text: string;
+}
+
+type RoadmapDataItem = {
+  text: string;
+  startDate: string;
+  targetDate: string;
+  status: string;
+  issueUrl: string;
+};
+
+interface CardViewProps {
+  sortedRoadmapData: RoadmapData[];
+  searchQuery: string;
+}
+
+function sortRoadmapDataByDate(roadmapData: RoadmapData[]): RoadmapData[] {
+  const sortedData = [...roadmapData];
+  sortedData.sort((a, b) => {
+    const startDateA = new Date(a.targetDate);
+    const startDateB = new Date(b.targetDate);
+
+    if (isNaN(startDateA.getTime())) {
+      return 1; // Item A doesn't have a valid date, so it should be placed below item B
+    } else if (isNaN(startDateB.getTime())) {
+      return -1; // Item B doesn't have a valid date, so it should be placed below item A
+    } else {
+      return startDateA.getTime() - startDateB.getTime();
+    }
+  });
+  return sortedData;
+}
+
+function formatDate(date: Date): string {
```

Comment:
I can see this being reused, maybe we could move it to utils

Reply:
done !